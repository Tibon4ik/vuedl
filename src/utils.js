/*
 * vuedl
 *
 * (c) Savaryn Yaroslav <yariksav@gmail.com>
 *
 * Some functions was imported from nuxt.js/lib/app/utils.js
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
*/

export function destroyVueElement (vm) {
  if (vm && !vm._isDestroyed && (typeof vm.$destroy === 'function')) {
    vm.$destroy()
  }
}

export function findContainer (container) {
  let found
  if (typeof container === 'string') {
    found = document.querySelector(container)
  } else {
    found = container
  }
  if (!found) {
    found = document.body
  }
  return found
}

// todo
// export function middlewareSeries (promises, appContext) {
//   if (!promises.length || appContext._redirected || appContext._errored) {
//     return Promise.resolve()
//   }
//   return promisify(promises[0], appContext)
//     .then(() => {
//       return middlewareSeries(promises.slice(1), appContext)
//     })
// }
