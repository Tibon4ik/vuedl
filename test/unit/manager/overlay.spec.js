// import Dialog from '../../../src/dialog'
import DialogManager from '../../../src/manager'
// import Confirm from '../../../src/components/Confirm'
// import { wrap } from '../../utils'
// import Returnable from '../../../src/mixins/returnable'
import Vue from 'vue'
import { sleep } from '../../utils'

describe('manager', () => {
  let manager

  // const asyncData = jest.fn((ctx) => {
  //   return sleep(10)
  // })

  const overlay = {
    template: '<div v-if="visible" id="overlay"/>',
    data () {
      return {
        visible: false
      }
    }
  }

  test('Check overlay correct work', async () => {
    manager = new DialogManager({ context: { store: {} } })
    manager.overlay('default', overlay)
    let dialog = manager.show({
      template: '<p/>',
      asyncData: sleep(10)
    })
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
    dialog = await dialog
    expect(dialog.element).toMatchSnapshot()
    dialog.close()
    await Vue.nextTick()
    expect(document.body.innerHTML).toMatchSnapshot()
  })
})
