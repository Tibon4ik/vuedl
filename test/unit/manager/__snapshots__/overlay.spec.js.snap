// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Check overlay correct work 1`] = `"<div id=\\"overlay\\"></div>"`;

exports[`manager Check overlay correct work 2`] = `
<div
  class="dialog-layout"
>
  <p />
</div>
`;

exports[`manager Check overlay correct work 3`] = `"<!---->"`;
